pkg/autoscaler/autoscaler.go

```diff
@@ -254,295 +256,406 @@ var _ = Describe("[Feature:Machines] Autoscaler should", func() {
+			By("Getting existing machinesets")
+			existingMachineSets, err := framework.GetMachineSets(client)
+			Expect(err).NotTo(HaveOccurred())
+			Expect(len(existingMachineSets)).To(BeNumerically(">=", 1))
 
-		By("Getting existing nodes")
-		existingNodes, err := framework.GetNodes(client)
-		Expect(err).NotTo(HaveOccurred())
-		Expect(len(existingNodes)).To(BeNumerically(">=", 1))
-
-		glog.Infof("Have %v existing machinesets", len(existingMachineSets))
-		glog.Infof("Have %v existing machines", len(existingMachines))
-		glog.Infof("Have %v existing nodes", len(existingNodes))
-		Expect(len(existingNodes) == len(existingMachines)).To(BeTrue())
-
-		// The remainder of the logic in this test requires 3
-		// machinesets.
-		var machineSets [3]*mapiv1beta1.MachineSet
-
-		randomUUID := string(uuid.NewUUID())
-		for i := 0; i < len(machineSets); i++ {
-			targetMachineSet := existingMachineSets[i%len(existingMachineSets)]
-			machineSetName := fmt.Sprintf("e2e-%s-w-%d", randomUUID[:5], i)
-			machineSets[i] = framework.NewMachineSet(targetMachineSet.Labels[framework.ClusterKey],
-				targetMachineSet.Namespace,
-				machineSetName,
-				targetMachineSet.Spec.Selector.MatchLabels,
-				targetMachineSet.Spec.Template.ObjectMeta.Labels,
-				&targetMachineSet.Spec.Template.Spec.ProviderSpec,
-				1) // one replica
-			machineSets[i].Spec.Template.Spec.Labels = map[string]string{
-				autoscalerWorkerNodeRoleLabel: "",
-			}
-			Expect(client.Create(ctx, machineSets[i])).Should(Succeed())
-			cleanupObjects[machineSets[i].Name] = machineSets[i]
-		}
+			By("Getting existing machines")
+			existingMachines, err := framework.GetMachines(client)
+			Expect(err).NotTo(HaveOccurred())
+			Expect(len(existingMachines)).To(BeNumerically(">=", 1))
 
-		By(fmt.Sprintf("Creating %v transient machinesets", len(machineSets)))
-		testDuration := time.Now().Add(time.Duration(framework.WaitLong))
-		Eventually(func() bool {
-			By(fmt.Sprintf("[%s remaining] Waiting for nodes to be Ready in %v transient machinesets",
-				remaining(testDuration), len(machineSets)))
-			var allNewNodes []*corev1.Node
+			By("Getting existing nodes")
+			existingNodes, err := framework.GetNodes(client)
+			Expect(err).NotTo(HaveOccurred())
+			Expect(len(existingNodes)).To(BeNumerically(">=", 1))
+
+			glog.Infof("Have %v existing machinesets", len(existingMachineSets))
+			glog.Infof("Have %v existing machines", len(existingMachines))
+			glog.Infof("Have %v existing nodes", len(existingNodes))
+			Expect(len(existingNodes) == len(existingMachines)).To(BeTrue())
+
+			// The remainder of the logic in this test requires 3
+			// machinesets.
+			var machineSets [3]*mapiv1beta1.MachineSet
+
+			randomUUID := string(uuid.NewUUID())
 			for i := 0; i < len(machineSets); i++ {
-				nodes, err := framework.GetNodesFromMachineSet(client, machineSets[i])
-				if err != nil {
-					return false
+				targetMachineSet := existingMachineSets[i%len(existingMachineSets)]
```

Comment:
What's the benefit of the module operation on the `existingMachineSets`? Would it make more sense to do `for i, targetMachineSet := range machineSets` to get `i` and `targetMachineSet`?